From 60189143a4097c615f9cda0dbec2cd48b4eca77d Mon Sep 17 00:00:00 2001
From: Mike Gorse <mgorse@suse.com>
Date: Wed, 30 Oct 2013 20:42:09 +0000
Subject: Print to a file in the current directory by default

When printing to a file, the filename was not being propagated if a
directory was not specified.

https://bugzilla.gnome.org/show_bug.cgi?id=711177
---
diff --git a/gtk/gtkprinteroptionwidget.c b/gtk/gtkprinteroptionwidget.c
index 2f43edf..08c2aff 100644
--- a/gtk/gtkprinteroptionwidget.c
+++ b/gtk/gtkprinteroptionwidget.c
@@ -521,7 +521,16 @@ filesave_changed_cb (GtkWidget              *button,
       if (g_uri_parse_scheme (file) != NULL)
         uri = g_strdup (file);
       else
-        uri = g_build_path ("/", gtk_file_chooser_get_uri (GTK_FILE_CHOOSER (priv->combo)), file, NULL);
+        {
+          gchar *chooser_uri = gtk_file_chooser_get_uri (GTK_FILE_CHOOSER (priv->combo));
+          if (chooser_uri)
+            {
+              uri = g_build_path ("/", chooser_uri, file, NULL);
+              g_free (chooser_uri);
+            }
+          else
+            uri = g_filename_to_uri (file, NULL, NULL);
+        }
     }
  
   if (uri)
--
cgit v0.9.2

